### DISCUSSION TITLE
Adding a baseline model(s)

### DISCUSSION THREAD

**Discussion by IsaacMacarthur** [reichlab/variant-nowcast-hub#313] [Ideas]
One suggestion that has come up in discussions on how to improve the hub is the idea of adding a baseline model(s) for comparison.  The scoring metric that the hub is currently using, the energy score, can be more easily interpreted if there is a baseline model with baseline scores that other models can be compared against.

We had one suggestion for a possible baseline model that would be fit using the following steps:
1.  We find the national proportion averages by variant over the last n weeks, for some n.
2.  Find the changes in the national proportion over the last m weeks by variant, and calculate the empirical distribution of the differences.
3. For every state, to generate one forecasted sample H days into the future for one variant, take the national proportion average and add H draws from the empirical distribution of the differences. We can then softmax over the variants to ensure they sum to 1. 

We would appreciate any feedback, both on the baseline model suggested here and on other possible baseline models that would be useful.

**Comment by elray1** [comment]
Here are two variations another idea with a different flavor:

 1. Fit a multinomial logistic regression to the total aggregated reported counts at the national level over the past $n$ days using maximum likelihood (i.e., frequentist methods to get point estimates).  Predictions will be the same for all states.
     1. Variation 1 is a standard MLR model in time: $c_t | \pi_t, n_t \sim Multinomial(\pi_t, n_t)$ where $\pi_{t} = \text{softmax}(\gamma_t)$ and $\gamma_{t,v} = \beta_{0,v} + \beta_{1,v} t$, with separate coefficients for each variant $v$.
     2. Variation 2 throws away the slope: $c_t | \pi_t, n_t \sim Multinomial(\pi_t, n_t)$ where $\pi_{t} = \text{softmax}(\gamma_t)$ and $\gamma_{t,v} = \beta_{0,v}$, with separate intercepts for each variant $v$
 2. Point predictions are obtained by getting predictions from this model
 3. Probabilistic sample predictions are obtained by somehow bootstrapping prediction errors that the models under point 1 would have had in past weeks.  Haven't thought through how to deal with the multinomial sampling bit.

**Comment by dylanhmorris** [comment]
Recording my suggestion from our f2f discussion that the error distribution be chosen to optimize retrospective coverage.

**Comment by IsaacMacarthur** [comment]
Per our discussion from today, here is the idea for a baseline model:
This builds off of Evan's idea 1 from above. First fit a standard MLR model in time, $c_t  | \pi_t,n_t  \sim multinomial(\pi_t,n_t)$ and $\pi_t= \text{softmax}(\gamma_t)$ for $\gamma_{(t,v)}= \beta_{(0,v)}+ \beta_{(1,v)} t$ to the national level data, using the last $m$ days of data before the target date. For my implementation, I have chosen $m = 42$. This gives us the mean predictions.
To obtain sample predictions, first find the empirical distribution of differences in weekly proportion by clade, i.e find the proportion of clade A in week $i + 1$ and subtract the proportion of clade A in week $i$, for all $i = 1,... m/7$. If any week has no data, report 0 instead.
Next to obtain one sample submission $i$, for one state $s$, for one nowcast date $t$, for one clade $v$, denoted $\theta_{(s,t,v)}^i$ take the mean prediction for that date and clade and add three draws with replacement from the empirical distribution of differences in weekly proportion for that clade.  If this results in a number less than 0, assign 0 for that clade instead. Repeat this process for each $v$, then $\theta_{(s,t)}^i =  \text{softmax}(\theta_{(s,t,v)}^i)$.
If we are forecasting instead of adding three draws from the empirical distribution of differences in weekly proportion for that clade, add three plus the number of days we are forecasting into the future to account for the greater uncertainty. 
The choice of the number of days of data to fit to and the number of draws to add from the empirical distribution were both chosen arbitrarily and could be refined.

**Reply by nickreich** [reply]
> For my implementation, I have chosen $m=42$. This gives us the mean predictions.

Did you experiment with other choices of $m$? I suggest using maybe $m=100$, so we have some longer history of more complete observations. If we find during some initial runs that recent incomplete days of observations are throwing off the model, we could truncate recent weeks of data as well.

> ...the empirical distribution of differences in weekly proportion by clade...

shouldn't this be by day? I'm confused about the timescale here, and then for each sample you are adding three draws with replacement? Can you explain this more? Why not just do empirical distribution of daily differences and add one per day?

> ... in week $i+1$...

Can we just use day $t$, so we don't confuse our temporal notation with the sample $i$?

**Comment by trobacker** [comment]
After a variant-nowcast modeling meeting, we've come to a conclusion on the first pass of a baseline model.

> Baseline model: Fit a Bayesian Multinomial Logistic Regression model with a linear logit-space growth at the national level, get uncertainty estimates and use those for each location (state). 

We may make adjustments to this baseline model in the future.
